operator/internal/central/extensions/reconcile_tls.go

```diff
@@ -218,6 +275,103 @@ func validateOrGenerateCA(oldCA mtls.CA, oldFileMap types.SecretDataMap) (mtls.C
 	return nil, nil, fmt.Errorf(msg, mtls.CACertFileName, mtls.CAKeyFileName)
 }
 
+func validateSecondaryCA(oldFileMap, newFileMap types.SecretDataMap) error {
+	caCert, caCertPresent := oldFileMap[mtls.SecondaryCACertFileName]
+	caKey, caKeyPresent := oldFileMap[mtls.SecondaryCAKeyFileName]
+
+	if caCertPresent && caKeyPresent {
+		_, err := certgen.LoadSecondaryCAFromFileMap(oldFileMap)
+		if err != nil {
+			// Secured Clusters might already be using the secondary CA to connect to Central, so re-creating it will
+			// not fix things.
+			return errors.Wrap(err, "invalid secondary CA in the existing secret, please delete it to allow re-generation")
+		}
+
+		newFileMap[mtls.SecondaryCACertFileName] = caCert
+		newFileMap[mtls.SecondaryCAKeyFileName] = caKey
+		return nil
+	} else if caCertPresent || caKeyPresent {
+		const msg = "malformed secret (%s present but %s missing), please delete it to allow re-generation"
+		if !caCertPresent {
+			return fmt.Errorf(msg, mtls.CAKeyFileName, mtls.CACertFileName)
+		}
+		return fmt.Errorf(msg, mtls.CACertFileName, mtls.CAKeyFileName)
+	}
+
+	return nil
+}
+
+func (r *createCentralTLSExtensionRun) checkCARotation(primary, secondary *x509.Certificate) (CARotationAction, error) {
+	if primary == nil {
+		return CARotateNoAction, errors.New("primary CA must not be nil")
+	}
+
+	startTime := primary.NotBefore
+	endTime := primary.NotAfter
+	if !endTime.After(startTime) {
+		return CARotateNoAction, fmt.Errorf("certificate expires at %s before it begins to be valid at %s", endTime, startTime)
+	}
+	if r.currentTime.Before(startTime) {
+		return CARotateNoAction, fmt.Errorf("certificate lifetime start %s is in the future", startTime)
+	}
+	if r.currentTime.After(endTime) {
+		return CARotateNoAction, fmt.Errorf("certificate expired at %s", endTime)
```

Comment:
These don't mention which certificate. But if the caller wraps the error message with one that does mention primary CA then they will be fine.